forem/forem
# Fastly edge caching: use PURGE HTTP method instead of POST

## What type of PR is this? (check all applicable)


## Description

This PR adds a feature flag called `fastly_http_purge` that we can use to enable the `PURGE` HTTP method to [purge an individual URL](https://github.com/fastly/fastly-ruby#efficient-purging).

Right now we're using the deprecated "purge by `POST`", we'd like to switch to the new method.

## Related Tickets & Documents

Closes https://github.com/forem/forem/issues/12326

## QA Instructions, Screenshots, Recordings

It's not possible to test this locally, I think we should deploy to https://community.benhalpern.com/ which uses Fastly, enable the flag in the `/admin/feature_flags/features` UI and then excersise caching to make sure stuff is purged.

We can use the `Fastly-Debug` header to test it, for example:

```shell
curl -H "Fastly-Debug: 1" -I https://community.benhalpern.com/{path}
```

## Added tests?

- [x] Yes (also: automatic tests for Fastly are moot as they can't be run locally)
      have not been included_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] [Storybook](https://storybook.forem.com/) (for Crayons components)
